website/pages/docs/nia/network-drivers.mdx

```diff
@@ -6,10 +6,49 @@ description: >-
+
+Within the Consul-Terraform-Sync configuration for a task, practitioners can select the desired module to run for the task as well the set of services for the module to execute on. Each task executed by the Terraform driver corresponds to an automated root module that calls the selected module in an isolated Terraform environment. Consul-Terraform-Sync will manage concurrent execution of these tasks.
+
+### Root Module
+
+The root module is simple in structure and proxies Consul service information, configuration, and other variables to the module. The content of the files that make up the root module are sourced from Consul Terraform Sync configuration, information for task's module to use as the automation playbook, and the Consul Catalog for discovering service information.
+
+Autogenerated root modules for tasks are stored in local subdirectories of the Terraform working directory. By default, the working directory `sync-tasks` is created in the current directory. To configure where Terraform configuration files are stored, set [`working_dir`](/docs/nia/installation/configuration#working_dir) for the Terraform driver to the desired path.
+
+```
+sync-tasks/
+  my-task/
+    main.tf
+    variables.tf
+    terraform.tfvars
+    terraform.tfvars.tmpl
+```
+
+The following files of the root module are generated for each task. An [example of a root module created by Consul-Terraform-Sync](https://github.com/hashicorp/consul-terraform-sync/tree/master/examples) can be found in the project repository.
+* `main.tf` - The main file contains the terraform block, provider blocks, and a module block calling the module configured for the task.
+    * `terraform` block - The corresponding provider source and versions for the task from the configuration files are placed into this block for the root module. The Terraform backend from the configuration is also templated here.
+    * `provider` blocks - The provider blocks generated in the root module resemble the provider blocks in the configuration. They have identical arguments present nad are set from the intermediate variable created per provider.
+    * `module` block - The module block is where the task's module is called as a [child module](https://www.terraform.io/docs/configuration/modules.html#calling-a-child-module). Required and optional input variables are passed as arguments to the module.
+* `variables.tf` - This file contains 2 types of variable defintions. The required `services` input variable which determines module compatibility with Consul-Terraform Sync (read more on [compatible Terraform modules](#compatible-terraform-modules) for more details) and various intermediate variables used to dynamically configure providers. These intermediate provider variables are interpolated from the provider blocks and arguments configured in the Consul-Terraform-Sync configuration.
```

Comment:
suggestion to change: "defintions" => "definitions"

Reply:
🙇